shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/Alert.kt

```diff
@@ -278,15 +360,24 @@ data class Alert(
         ): List<Alert> {
             val stopIds = trip.stopIds ?: emptyList()
 
-            val alerts = alerts.filter { it.hasStopsSpecified }
+            val alerts =
+                alerts.filter {
+                    it.hasStopsSpecified && it.significance >= AlertSignificance.Accessibility
+                }
 
             val targetStopAlertIds =
-                applicableAlerts(
-                        alerts.toList(),
-                        trip.directionId,
-                        listOf(trip.routeId),
-                        targetStopWithChildren
-                    )
+                alerts
+                    .filter {
+                        it.anyInformedEntitySatisfies {
+                            checkActivityIn(
+                                InformedEntity.Activity.Exit,
+                                InformedEntity.Activity.Ride
+                            )
+                            checkDirection(trip.directionId)
+                            checkRoute(trip.routeId)
+                            checkStopIn(targetStopWithChildren)
+                        }
+                    }
```

Comment:
The idea with the `PredicateBuilder` was to get the clarity of `applicableAlerts` with enough flexibility to require different activities instead in the context of downstream alerts.